### DISCUSSION TITLE
Question about the EdDSA example on the play ground.

### DISCUSSION THREAD

**Discussion by beegmon** [Consensys/gnark#502] [General]
I am trying to tweak the playground example EdDSA to use compressed PublicKey and Signatures as circuits inputs vs the present X/Y, an X/Y/S inputs.

I am also very new to go in general so that probably isn't helping me either here.

Looking at eddsa.PublicKey/Signature there are helper functions that take in a buf that is a compressed version of the input and sets X/Y and X/Y/S.

```
type Circuit struct {
	PublicKey eddsa.PublicKey   `gnark:",public"`
	Signature eddsa.Signature   `gnark:",public"`
	Message   frontend.Variable `gnark:",public"`
}
```

That is struct used the example and I assumed (wrongly) I could do something like:

```
type Circuit struct {
	PublicKey eddsa.PublicKey.Assign(tewards.BN254, PublicKey)   `gnark:",public"`
	Signature eddsa.Signature .Assign(tewards.BN254, Signature)  `gnark:",public"`
	Message   frontend.Variable `gnark:",public"`
}
```

That is very obviously wrong but I am struggling on how to figure out how to convert my input which is compressed in a hex string to th needed X/Y and X/Y/R values to feed into the circuit example.

**Comment by ivokub** [comment]
So, if you want to define the witness in the playground then in the end of the text input area there is a JSON object which you can use to submit the witness. The values in the playground must already be deserialized into primitive native field elements.
<img width="967" alt="image" src="https://user-images.githubusercontent.com/1309050/221996303-a0d45213-de53-4454-9bec-cd55cbaa1a9b.png">

Usually in the application which integrates gnark you can do pre-processing to obtain the required format. Playground doesn't provide this option for simplicity.
